### PR TITLE
[PATCH v2] linux-gen: check odp_cpu_id() call return values

### PR DESCRIPTION
**Pull request by MatiasElo**
Check odp_cpu_id() call return values to prevent accidentally using negative return values as array indices. Fixes Coverity warnings.

### CODE REVIEW

File: platform/linux-generic/odp_schedule_scalable.c [hunk 1]
```diff
@@ -927,6 +927,9 @@ static int _schedule(odp_queue_t *from, odp_event_t ev[], int num_evts)
 	}
 
 	cpu_id = odp_cpu_id();
+	if (odp_unlikely(cpu_id < 0))
```
Comment (JannePeltonen) [inline platform/linux-generic/odp_schedule_scalable.c:930]: The cpu ID here should perhaps be thread ID, but that is orthogonal to this change, which looks ok.

**Review by JannePeltonen — APPROVED** [review]

File: platform/linux-generic/odp_schedule_scalable.c (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-generic/odp_schedule_scalable.c:930]: Yeah, I was thinking the same, but didn't want to touch the scalable scheduler code any more than necessary.
